cli/kafka/config.go

```diff
@@ -70,46 +135,64 @@ func ValidateConfig(c Config) error {
 				if c.Fields[i].Name == "" {
 					return errors.Errorf("a name attribute (which isn't equal to \"\") should exist for all fields")
 				}
+				if c.Fields[i].SourceType == "" {
+					return errors.Errorf("a source-type attribute (which isn't equal to \"\") should exist for all fields")
+				}
 			}
-			if found != 1 {
-				return errors.Errorf("exactly one primary key field is required")
+			if found < 1 {
+				return errors.Errorf("at least one primary key field is required")
 			}
 		}
 	}
+
+	return nil
+}
+
+// Only primary key fields required
+func validateConfigAvro(c Config) error {
 	return nil
 }
 
 // ConvertConfig converts a Config to one that suitable for IDK.
 func ConvertConfig(c Config) (ConfigForIDK, error) {
-	// Set a default kafka host in case one isn't provided.
-	hosts := []string{"localhost:9092"}
-	if len(c.Hosts) > 0 {
-		hosts = c.Hosts
-	}
 
 	// Copy all the shared members from Config to ConfigForIDK.
 	out := ConfigForIDK{
-		Hosts:             hosts,
+		Hosts:             c.Hosts,
 		Group:             c.Group,
 		Topics:            c.Topics,
 		BatchSize:         c.BatchSize,
 		BatchMaxStaleness: c.BatchMaxStaleness,
 		Timeout:           c.Timeout,
 		Table:             c.Table,
+		Encode:            c.Encode,
 	}
 
 	if len(c.Fields) == 0 {
 		return out, errors.New("fields cannot be empty")
 	}
 
-	// rawFields wil be the same as c.Fields, but possibly enhanced.
+	// rawFields will be the same as c.Fields, but possibly enhanced.
 	rawFields := make([]idk.RawField, 0, len(c.Fields))
 
-	var foundPK bool
+	var stringKeys bool
+	primaryKeys := []string{}
 	for _, fld := range c.Fields {
+
+		// handle primary key
 		if fld.PrimaryKey {
-			out.IDField = fld.Name
-			foundPK = true
+			switch keyType := fld.SourceType; keyType {
+			case dax.BaseTypeID:
+				// no-op
+			case dax.BaseTypeIDSet, dax.BaseTypeStringSet, dax.BaseTypeIDSetQ, dax.BaseTypeStringSetQ:
+				// key should be field that cannot contain multiple values
+				return out, errors.Errorf("Invalid")
+			default:
```

Comment:
I think I would be explicit about which types _can_ be used as string primary keys, and have the default case return the error (rather than being explicit about which _can't_).

Reply:
Yep, think this makes sense to me.